### PR TITLE
Make migration 30 idempotent

### PR DESCRIPTION
**Pull request by paulmooring**
NOT READY YET

This changes the `mover_actor_keys_access_group_callback` migration to not fail on multiple runs for a migration that bypasses partybus to avoid stopping all frontend services.

### CODE REVIEW

**Comment by marcparadise** [comment]
@paulmooring any plans to go forward with this?

File: src/chef-mover/src/mover_actor_keys_access_group_callback.erl [hunk 1]
```diff
@@ -134,21 +137,28 @@ add_group_to_group(#group{authz_id = IdToAdd}, TargetId) ->
...
+                {ok, _} ->
+                    AuthzId;
+                {error, _} ->
+                    {error, failure_creating_group}
+            end;
+        {ok, [AuthzId | _]} ->
             AuthzId;
-        {error, _} ->
+        _ ->
             {error, failure_creating_group}
```
Comment (stevendanna) [inline src/chef-mover/src/mover_actor_keys_access_group_callback.erl]: This failure message is a bit inaccurate now. Perhaps use: `{error, {group_query_failure, Error}}`

File: src/chef-mover/src/mover_actor_keys_access_group_callback.erl [hunk 2]
```diff
@@ -134,21 +137,28 @@ add_group_to_group(#group{authz_id = IdToAdd}, TargetId) ->
...
-                                              AuthzId,
-                                              Name,
-                                              SuperuserId,
-                                              Now,
-                                              Now
-                                             ]) of
-        {ok, _} ->
+    case sqerl:select(select_group_query(), [Name, OrgId], rows_as_scalars, [authz_id]) of
+        {ok, none} ->
+            Now = os:timestamp(),
```
Comment (stevendanna) [inline src/chef-mover/src/mover_actor_keys_access_group_callback.erl]: Maybe move 142-158 into a function.

**Comment by stevendanna** [comment]
@paulmooring Let's chat about what the outstanding issues are surrounding migrations in Hosted.  I want to make sure we have a clear picture of where we are, what migrations need work, and the timeline we want to get that work done over.

**Comment by stevendanna** [comment]
@paulmooring I'm going to close this.  If we find problems with migration 30 during your testing, we'll open a new fix for it at that point.

**Pull request closed without merging**